clang/lib/CConv/Utils.cpp

```diff
@@ -259,3 +259,14 @@ bool canWrite(const std::string &FilePath) {
   getAbsoluteFilePath(FilePath, fileAbsPath);
   return fileAbsPath.rfind(BaseDir, 0) == 0;
 }
+
+unsigned longestCommonSubsequence(const char *Str1, const char *Str2,
+                                  unsigned Str1Len, unsigned Str2Len) {
```

Comment:
The time complexity of the recursive LCS is `O(2^(m+n))`. Could you please change this to a memoized version which has a complexity of `O(mn)`?